packages/core/core/src/requests/AssetGraphRequest.js

```diff
@@ -201,6 +202,20 @@ export class AssetGraphBuilder {
 
     dumpToGraphViz(this.assetGraph, 'AssetGraph');
 
+    for (let asset of this.assetGraph.nodes.entries()) {
```

Comment:
I don't think we should be iterating every node in the graph to do this. Can environments be normalized as they're created?